### PR TITLE
updated tests

### PR DESCRIPTION
**Pull request by suryam0801**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: core/jazz_cloud-logs-streamer/test/test.js [hunk 1]
```diff
@@ -16,12 +16,31 @@
 
 const assert = require('chai').assert;
 const index = require('../index');
+const awsContext = require('aws-lambda-mock-context');
+const StreamZip = require('node-stream-zip');
 
+const zip =
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/test/test.js]: What is this line for?

File: core/jazz_cloud-logs-streamer/index.js [hunk 2]
```diff
@@ -65,6 +65,7 @@ module.exports.handler = (input, context, cb) => {
 
                 if (failedItems && failedItems.length > 0) {
                     logger.error("Failed Items: " + JSON.stringify(failedItems, null, 2));
+                    return cb(JSON.stringify(errorHandler.throwInputValidationError("101", "Failed Items")));
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/index.js]: This isn’t a micro service so there is no point in returning a error code. Moreover line 71 will never run now

File: core/jazz_cloud-logs-streamer/package.json [hunk 3]
```diff
@@ -6,13 +6,16 @@
     "private": true,
     "author": [],
     "scripts": {
+        "start": "node index",
         "test": "mocha"
     },
     "dependencies": {
-        "unicode-byte-truncate": "*"
+        "node-stream-zip": "^1.8.2",
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/package.json]: Shouldn’t this be devdependency?

File: core/jazz_login/package.json [hunk 4]
```diff
@@ -6,18 +6,22 @@
     "private": true,
     "author": [],
     "scripts": {
+        "start": "node index",
```
Comment (suryajak) [inline core/jazz_login/package.json]: Remove everything about login from this PR

**Review by suryajak — CHANGES_REQUESTED** [review]

File: core/jazz_cloud-logs-streamer/index.js [hunk 5]
```diff
@@ -41,8 +41,9 @@ module.exports.handler = (input, context, cb) => {
 
     // decompress the input
     zlib.gunzip(zippedInput, function (error, buffer) {
-        if (error) { context.fail(error); return; }
-
+        if (error) { context.fail(error); 
+            return callback(JSON.stringify(errorHandler.throwNotFoundError("100", "Context Failed Error")));
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/index.js]: This PR is still treating this as a microservice. Don't throw "http" errors as there is no client.

File: core/jazz_login/package.json (hunk 4 above)
Comment (suryajak) [reply core/jazz_login/package.json]: Can you remove Login related stuff from this PR?

**Review by suryajak — CHANGES_REQUESTED** [review]

File: core/jazz_cloud-logs-streamer/components/utils.js [hunk 6]
```diff
@@ -52,6 +52,7 @@ var getInfo = function (messages, patternStr) {
 	if (messages) {
 		for (var i = 0, len = messages.length; i < len; i++) {
 			var _tmp = pattern.exec(messages[i].message);
+			console.log(messages[i].message)
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/components/utils.js]: Remove this debug line

**Review by suryajak — CHANGES_REQUESTED** [review]

**Comment by suryajak** [comment]
There are conflicts in your PR, please resolve them.

File: core/jazz_cloud-logs-streamer/package.json (hunk 3 above)
Comment (suryajak) [reply core/jazz_cloud-logs-streamer/package.json]: I don't see any change but the comment was resolved.

File: core/jazz_cloud-logs-streamer/test/test.js [hunk 7]
```diff
@@ -14,14 +14,46 @@
...
-const index = require('../index');
+const expect = require('chai').expect;
+const utils = require("../components/utils");
+const index = require("../index");
 
-describe('Sample', function () {
-    it('tests handler', function (done) {
+describe('jazz_cloud-logs-streamer', function () {
+    describe('utils', () => {
+        describe('isNumeric', function () {
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/test/test.js:226]: Negative test cases are missing

File: core/jazz_login/package.json (hunk 4 above)
Comment (suryajak) [reply core/jazz_login/package.json]: This has been marked resolved but I don't see the changes. What am I missing?

**Review by suryajak — CHANGES_REQUESTED** [review]
There are open comments as well as merge conflicts, please take care of them.

File: core/jazz_cloud-logs-streamer/package.json (hunk 3 above)
Comment (suryam0801) [reply core/jazz_cloud-logs-streamer/package.json]: removing this line results in not being able to start the node package manager

Comment (suryajak) [reply core/jazz_cloud-logs-streamer/package.json]: The comment is about line 12+

File: core/jazz_cloud-logs-streamer/index.js [hunk 8]
```diff
@@ -83,7 +83,7 @@ function transform(payload) {
     if (payload.messageType === 'CONTROL_MESSAGE') {
         return null;
     }
-    logger.info("Raw payload..:" + JSON.stringify(payload));
+    //logger.info("Raw payload..:" + JSON.stringify(payload));
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/index.js]: Any reason we are having to change the Log lines in actual code for writing unit tests?

File: core/jazz_cloud-logs-streamer/test/test.js [hunk 9]
```diff
@@ -14,14 +14,180 @@
...
+describe('jazz_cloud-logs-streamer', function () {
+    describe('utils', () => {
+        // Tests for the isNumeric private function in utils.js
+        describe('isNumeric', function () {
+            /* Given any value, isNumeric will determine whether the value is numeric or not
+            * @params {Object} any value (string, int, float, array) that contains data
+            * @returns {Boolean} true if passed parameter is numeric, false if it is not numeric
+            */
+            // positive test for isNumeric Function
+            it('should indicate a valid numeric value', function () {
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/test/test.js]: There are other unit test cases for integers, negative numbers.

File: core/jazz_cloud-logs-streamer/test/test.js [hunk 10]
```diff
@@ -14,14 +14,180 @@
...
+            @returns {Boolean} true if given input string is a valid json file , false if given input string is not a valid json file
+            */
+
+            //positive test for isValidJson
+            it('should indicate a valid JSON', function () {
+                var json = '{ "Version": "2012-10-17","Statement": [{"Sid": "","Effect": "Allow","Principal": {"Service": "cloudtrail.amazonaws.com"},"Action": "sts:AssumeRole"}]}';
+                expect(utils().isValidJson(json)).to.equal(true);
+            });
+            it('should indicate a invalid JSON', function () {
+                var json = '{ "Version" "2012-10-17","Statement": [{"Sid": "","Effect": "Allow","Principal": {"Service": "cloudtrail.amazonaws.com"},"Action": "sts:AssumeRole"}]}';
```
Comment (suryajak) [inline core/jazz_cloud-logs-streamer/test/test.js]: It isn't pretty obvious what makes this a negative test case. Try making test criteria simple where possible. At the same time keep looking for the corner cases where the test setup and/or criteria might be complex.

**Review by suryajak — CHANGES_REQUESTED** [review]
Added a few comments. Also please remove everything about login service from this PR.

File: core/jazz_cloud-logs-streamer/package.json (hunk 3 above)
Comment (suryam0801) [reply core/jazz_cloud-logs-streamer/package.json]: removed the '"node-stream-zip": "^1.8.2"," from the dependency surya

Comment (suryam0801) [reply core/jazz_cloud-logs-streamer/package.json]: is that what you were looking for? i was not entirely sure, please let me know

File: core/jazz_cloud-logs-streamer/index.js (hunk 8 above)
Comment (suryam0801) [reply core/jazz_cloud-logs-streamer/index.js]: my bad surya. i was debugging and forgot to remove this line

File: core/jazz_cloud-logs-streamer/test/test.js (hunk 10 above)
Comment (suryam0801) [reply core/jazz_cloud-logs-streamer/test/test.js]: ok surya will do

**Review by suryam0801 — COMMENTED** [review]
accidental comment. meant to remove loginfile from PR

File: core/jazz_login/package.json [hunk 11]
```diff
@@ -1,23 +0,0 @@
-{
-    "name": "login",
```
Comment (suryajak) [inline core/jazz_login/package.json:2]: You need to remove all other micro services from this PR which includes the login service. But here you are deleting the file and not removing it from PR. Please take Mani's help here since we have been talking about this for sometime now.

**Review by suryajak — CHANGES_REQUESTED** [review]
